src/components/views/rooms/MessageComposer.tsx

```diff
@@ -426,7 +427,7 @@ export default class MessageComposer extends React.Component<IProps, IState> {
             />;
         }
 
-        const threadId = this.props.relation?.rel_type === RelationType.Thread
+        const threadId = this.props.relation?.rel_type === THREAD_RELATION_TYPE.name
```

Comment:
Shouldn't this use `THREAD_RELATION_TYPE.matches(relation?.rel_type)` to support both legacy and new?